src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotUseDeprecatedSecurityProtocolsTests.cs

```diff
@@ -96,6 +96,91 @@ public void TestMethod()
 }");
         }
 
+        [Fact]
+        public void TestUse192CompoundAssignmentDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Net;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        ServicePointManager.SecurityProtocol |= (SecurityProtocolType)192;
```

Comment:
What if the assignment target is a method call returning the required value, but one of the arguments to the method call has unsafe bits:
`ServicePointManager.SecurityProtocol |= SomeMethod(ignoredArg: (SecurityProtocolType)192);`

Reply:
False positive.  I was thinking that's probably an okay trade off, since that doesn't seem like a typical pattern, and there usually isn't a heck of a lot of code dealing SecurityProtocolTypes.

Reply:
Sounds good.

Reply:
If you wanted to be pedantic, you can add extra check by walking up the parent chain of found operation to only allow conversion and binary operations before you hit the assignment operation

Reply:
Though then you miss out catching conditional assignments and such . Given this is a security rule, I agree that better to have some false positives then missing on flagging real violations